docs/basics.rst

```diff
@@ -159,7 +159,7 @@ Python's ``setuptools`` to build the module, while `cmake_example`_ uses CMake
 
 Building the above C++ code will produce a binary module file that can be
 imported to Python. Assuming that the compiled module is located in the
-current directory (in newer version of python, rename the `example.cpython-xxx-linux-gun.so` to `example.so`) , the following interactive Python session shows how to
+current directory (in newer version of python, rename the ``example.cpython-xxx-linux-gun.so`` to ``example.so``) , the following interactive Python session shows how to
```

Comment:
Did you see example.cpython-*.so with python_example, or cmake_example, or both?

@henryiii could this be fixed in the setuptools and/or cmake config?

Reply:
You do not want to do this. This is not Python version specific - it's always worked and always been a bad idea. The SOABI that you are removing is how it avoids loading on a version of Python it wasn't compiled for. PyPy won't even load an extension without the SOABI.

Reply:
("this" meaning this PR, not this comment)

Is there some problem you are trying to solve? It would help to know what the issue is. Is some setup accedently producing `example.so` instead of a properly named extension? The pybind11 tools should do this correctly; if you use FindPython directly, you need to pass the SOABI flag (sadly not the default).

Reply:
Totally aside: there is a typo (not fixed here) in basics.rst of `linux-gun` -> `linux-gnu`

Reply:
Ah, yes, didn't see that. Pretty sure we aren't accepting this unless more info into why it's needed is given. Nothing should be producing binaries without SOABI, so maybe it's a fix somewhere else that's needed.